### ISSUE TITLE
Access to error handler

### ISSUE THREAD

**Issue by adlondon** [medipass/react-credit-card-input#24]
Thanks so much for this awesome library. Really appreciate all the hard work that's gone into it.

Im wondering if there is a way to have access to the valid / invalid state via an onChange callback of some kind. Looking to see if the entered card is valid, and if it is, allow the user to move on to the next part of the form.

**Comment by ayushkhanduri** [comment]
Is someone working on this ?

**Comment by jxom** [comment]
Not at the moment!

**Comment by mmmikeal** [comment]
Hi I am trying to add this to your repo because I need it for a project that I'm working on, but I am having a <s>bit</s> lot  of trouble following your conditional logic at the beginning of this function. As you can see, i am trying to add a second callback function handleError in the same way that you provide onChange but am really suffering integrating in your conditional logic that seems to check if the user entered an onChange...

I can see where I need to execute my callback function (` // executeHandleError() `)

```javascript



![screen shot 2018-12-14 at 3 06 26 pm](https://user-images.githubusercontent.com/13124444/50025113-7eb73e80-ffb2-11e8-8363-66fb6046d32c.png)

handleCardNumberChange = (
    { 
      onChange, 
      handleError 
    }: { onChange?: ?Function } = { onChange: null }
  ) => (e: SyntheticInputEvent<*>) => {
    const { customTextLabels, enableZipInput } = this.props;
    const cardNumber = e.target.value;
    const cardNumberLength = cardNumber.split(' ').join('').length;
    const cardType = payment.fns.cardType(cardNumber);
    const cardTypeInfo =
      creditCardType.getTypeInfo(creditCardType.types[CARD_TYPES[cardType]]) ||
      {};
    const cardTypeLengths = cardTypeInfo.lengths || [16];

    this.cardNumberField.value = formatCardNumber(cardNumber);

    this.setState({
      cardImage: images[cardType] || images.placeholder,
      cardNumber
    });

    if (enableZipInput) {
      this.setState({ showZip: cardNumberLength >= 6 });
    }

    this.setFieldValid();
    if (cardTypeLengths) {
      const lastCardTypeLength = cardTypeLengths[cardTypeLengths.length - 1];
      for (let length of cardTypeLengths) {
        if (
          length === cardNumberLength &&
          payment.fns.validateCardNumber(cardNumber)
        ) {
          // execute handleError()
          this.cardExpiryField.focus();
          break;
        }
        if (cardNumberLength === lastCardTypeLength) {
          this.setFieldInvalid(
            customTextLabels.invalidCardNumber || 'Card number is invalid'
          );
          // execute handleError()
        }
      }
    }
```

**Comment by mmmikeal** [comment]
`{ onChange?: ?Function } = { onChange: null }`

i really can't understand this syntax

**Comment by mmmikeal** [comment]
HAHA i figured it out, i dont think i need those at all.

**Comment by mmmikeal** [comment]
https://github.com/medipass/react-credit-card-input/pull/35

not expecting this to pass, but can someone with more experience with the codebase take a look at this? This feature seems key to the functionality of a credit card form

**Comment by jxom** [comment]
Fixed in `1.1.0` via #36
